pkg/daemon/update.go

```diff
@@ -601,13 +601,22 @@ func generateKargsCommand(oldConfig, newConfig *mcfgv1.MachineConfig) []string {
 	return cmdArgs
 }
 
+func isOSTreeBased(osName string) bool {
```

Comment:
The canonical way to do this is to check for `/run/ostree-booted`.

Reply:
I've removed that func entirely for now and just expanded on the way it's been done so far. Do you think this is sufficient?

Reply:
Not opposed to landing this patch as is, but I think it'd be a *lot* cleaner to add an `bool osIsCoreOS` to the daemon and conditionalize everything on that.

(`isOSTreeBased` is an intermediate level - for example it would also apply to e.g. RHEL Atomic Host, but that's not "CoreOS" which I think is now the pairing of (Ignition, OSTree) - we could still check that and there *might* be a place that would make sense to use that)

Reply:
I agree. Also @runcom for visibilty, this may be something for the MCO backlog? :)